src/indexing/strategies/seqnos_by_key_bucketed.cpp

```diff
@@ -124,6 +124,21 @@ namespace ccf::indexing::strategies
         "{}: {} -> {} for {}", name, range.first, range.second, hex_key);
     }
 
+    void store_empty_buckets(
+      const ccf::ByteVector& k, Range begin, const Range& end)
+    {
+      while (begin != end)
+      {
+        LOG_TRACE_FMT(
+          "Storing empty bucket for range [{}, {}) for key {:02x}",
+          begin.first,
+          begin.second,
+          fmt::join(k, ""));
+        store_to_disk(k, begin, {});
+        begin = get_range_for(begin.second);
+      }
+    }
```

Comment:
This is the main fix. The way this strategy works is to divide the history of seqnos into equal-sized ranges, and create `(range, kv-key)` keyed _buckets_ of results, which have a predictable name so can be easily fetched. For instance, we might write index chunks like the following, if the bucket-size was 10 (where the right-hand side content is "every time this key was written to"):

```
([0, 10), "hello") => {4, 7, 9}
([10, 20), "hello") => {10, 13, 14, 17}
([10, 20), "saluton") => {10, 11, 15}
([40, 50), "saluton") => {42}
```

This function makes sure that, in the case above, where we didn't receive any writes to key `"saluton"` before seqno 10, or between seqnos 20 and 40, we still produce appropriate empty chunks when we're aware of them:
```
([0, 10), "saluton") => {}
([20, 30), "saluton") => {}
([30, 40), "saluton") => {}
```